### DISCUSSION TITLE
supabase.auth.session() challenges - getting - Error: Auth session missing!

### DISCUSSION THREAD

**Discussion by fadethegap** [supabase/supabase#5875] [Questions] [answered]
I followed Jon Meyers [methodology](https://egghead.io/lessons/supabase-pass-supabase-session-cookie-to-api-route-to-identify-user) to implement a user object throughout my application. Everything worked for perhaps a couple of weeks, and then I started getting a 500 error when logging out.
![Screen Shot 2022-03-09 at 3 06 40 PM](https://user-images.githubusercontent.com/47059192/157525059-698e2bdb-d337-4b1b-a182-148b58e2851e.png)
![Screen Shot 2022-03-09 at 3 08 23 PM](https://user-images.githubusercontent.com/47059192/157525288-ca2b70f5-4f44-4ef1-b936-036c20fb23e8.png)

I forced a temporary fix by checking for the session prior setting a cookie:
![Screen Shot 2022-03-09 at 3 10 22 PM](https://user-images.githubusercontent.com/47059192/157525721-1c929112-2914-420a-9d0c-5e89853ee793.png)
Incidentally, that console.log does produce a session object...

That api call goes to:
![Screen Shot 2022-03-09 at 3 12 30 PM](https://user-images.githubusercontent.com/47059192/157527682-f77b24ce-1c84-4244-93d9-1bbd97657546.png)

I forgot about it until my Stripe calls stopped working due to a JWT expiry, and I was hoping they were related.
![Screen Shot 2022-03-09 at 3 19 00 PM](https://user-images.githubusercontent.com/47059192/157528233-144b21f7-5488-421d-ab95-830d6c0312f2.png)

I am returning a user from the getUserByCookie call:
![Screen Shot 2022-03-09 at 3 24 26 PM](https://user-images.githubusercontent.com/47059192/157529830-77210d31-e9c8-4f6b-8972-d3ee46fc9587.png)

But I'm getting the JWT expiration error when I try to access supabase.
![Screen Shot 2022-03-09 at 3 15 44 PM](https://user-images.githubusercontent.com/47059192/157527127-a29b3d72-6480-4180-9ff0-0826a1f5d1db.png)

I may be dealing with a couple of issues, but any guidance would be greatly appreciated.

**Comment by fadethegap** [comment, marked as answer]
Well, I'm a bonehead.

The first issue I raised above still vexes me, but my Stripe issue was a RLS change I had made.

Sorry for the noise.

**Reply by matthannam** [reply]
I'm also following John Meyers pattern on this and am bumping into the same problem.  500 error on logout.

**Comment by GBrachetta** [comment]
Same here, getting a 500 on logout using this pattern!
